### PR TITLE
update Cyclic Array

### PR DESCRIPTION
**Pull request by Sosuke23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC (Python): 
Screenshot from 2024-12-17 13-09-06
Screenshot from 2024-12-17 13-09-12

AC (C++):
Screenshot from 2024-12-17 13-09-22
Screenshot from 2024-12-17 13-09-29

### CODE REVIEW

File: solutions/platinum/cses-1191.mdx [hunk 1]
```diff
@@ -124,5 +124,102 @@ int main() {
 	printf("%d\n", r);
 }
 ```
+
 </CPPSection>
+<PySection>
+
+<Warning>
+
+Due to Python's constant factor, the below solution TLEs on a couple test cases.
```
Comment (dongliu0426) [inline solutions/platinum/cses-1191.mdx:133]: This problem is doable in $O(N\log N)$ by computing binary lifting pointers and for each $i$, jumping until it reaches $i+n$.

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/platinum/cses-1191.mdx (hunk 1 above)
Comment (Sosuke23) [reply solutions/platinum/cses-1191.mdx:133]: I can't see your solution, it's all blank

Comment (Sosuke23) [reply solutions/platinum/cses-1191.mdx]: I updated the c++ and py code along with the explanation for the updated solution which runs in `(n log n)` time.

**Review by dongliu0426 — APPROVED** [review]
Great!

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**